tfjs-converter/src/executor/graph_executor.ts

```diff
@@ -337,25 +374,32 @@ export class GraphExecutor implements FunctionExecutor {
   }
 
   disposeIntermediateTensors() {
-    if (!this.intermediateTensors) {
+    if (this.dumpMode === DumpMode.Default) {
       return;
     }
-    Object.keys(this.intermediateTensors)
-        .forEach(
-            key => this.intermediateTensors[key].forEach(
-                tensor => tensor.dispose()));
-    this.disposeTensorsMap();
+    if (this.tensorsPendingDisposal) {
+      this.tensorsPendingDisposal.forEach(tensor => {
+        tensor.dispose();
+      });
+      this.tensorsPendingDisposal = null;
+    }
+    if (this.dumpMode === DumpMode.Async) {
+      this.disposeTensorsMap();
```

Comment:
why we need to call `this.disposeTensorsMap()` only for async mode?

Reply:
This is intended behaviour: 
In sync mode,  intermediat tensors are kept in tensorsPendingDisposal.
In async mode, intermediat tensors = = tensormap - tensorsToKeep.

This is why we have two different dispose. 
Please note in the updated version, I tried to align these two path, which means, for both mode, all intermediat tensors are kept in tensorsPendingDisposal.